config/routes.rb

```diff
@@ -1,5 +1,9 @@
 Rails.application.routes.draw do
-  resources :posts
+  get 'advertisements/index'
+
+  get 'advertisements/show'
+
+  resources :posts, :advertisements
```

Comment:
This line should take care of defining `advertisements/index` and `advertisements/show` for you (in addition to the other RESTful actions :smile:). You can safely remove the two `get` routes defined above.

If you are interested in defining a RESTful resource with just the actions you want, you can do:

```ruby
resources :advertisements, only: [:index, :show]
```

You can also do the opposite filter:

```ruby
resources :advertisements, except: [:new, :create, :edit, :update, :destroy]
```

Keep in mind that you will need to define the `posts` resource separately (in a new line). Using `except` or `only` would affect all of the resources in that declaration, and we don't want that for `posts`. Your routes would look like then look like this:

```ruby
resources :posts
resources :advertisements, only: [:show, :index]
```